spec/controllers/admin/admin_users_controller_spec.rb

```diff
@@ -6,83 +6,87 @@
       end
     end
 
-    context "when admin has correct authorization" do
-      it "allows access to index" do
-        admin.update!(roles: ["policy_and_abuse"])
-        fake_login_admin(admin)
-        get :index
+    (Admin::VALID_ROLES - search_roles).each do |role|
+      context "with role #{role}" do
+        let(:admin) { create(:admin, roles: [role]) }
 
-        expect(response).to have_http_status(:success)
+        it "redirects with error" do
+          it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+        end
       end
     end
   end
 
-  describe "GET #bulk_search" do
-    let(:admin) { create(:admin) }
-
-    context "when admin does not have correct authorization" do
-      it "redirects with error" do
-        admin.update!(roles: [])
-        fake_login_admin(admin)
-        get :bulk_search
+  shared_examples "permits access to authorized admins" do
+    search_roles.each do |role|
+      context "with role #{role}" do
+        let(:admin) { create(:admin, roles: [role]) }
 
-        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+        it "allows access to index" do
+          expect(response).to have_http_status(:success)
+        end
       end
     end
+  end
 
-    context "when admin has correct authorization" do
-      it "allows access to access bulk search" do
-        admin.update!(roles: ["policy_and_abuse"])
-        fake_login_admin(admin)
-        get :bulk_search
+  describe "GET #index" do
+    before do
+      fake_login_admin(admin)
+      get :index
+    end
 
-        expect(response).to have_http_status(:success)
-      end
+    it_behaves_like "denies access to unauthorized admins"
+    it_behaves_like "permits access to authorized admins"
+  end
+
+  describe "GET #bulk_search" do
+    before do
+      fake_login_admin(admin)
+      get :bulk_search
     end
+
+    it_behaves_like "denies access to unauthorized admins"
+    it_behaves_like "permits access to authorized admins"
   end
 
   describe "GET #show" do
-    let(:admin) { create(:admin) }
     let(:user) { create(:user) }
 
-    context "when admin does not have correct authorization" do
-      it "redirects with error" do
-        admin.update!(roles: [])
-        fake_login_admin(admin)
-        get :show, params: { id: user.login }
-
-        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
-      end
+    before do
+      fake_login_admin(admin)
+      get :show, params: { id: user.login }
     end
 
+    it_behaves_like "denies access to unauthorized admins"
+    it_behaves_like "permits access to authorized admins"
+
     context "when admin has correct authorization" do
+      let(:admin) { create(:admin, roles: ["policy_and_abuse"]) }
```

Comment:
There is also this option:
```suggestion
      let(:admin) { create(:policy_and_abuse_admin) }
```